.github/ISSUE_TEMPLATE/release.md

```diff
@@ -0,0 +1,109 @@
+---
+name: Release
+about: Create a new release [for release managers only]
+title: 'Release MAJOR.MINOR.PATCH'
+labels: 'release'
+assignees: ''
+
+---
+
+Create a new release.
+
+**Summary**
+
+<--release summary to be used in announcements-->
+
+**Commit**
+
+<--latest commit ID to include in this release-->
+
+**Changelog**
+
+<--add notices from PRs merged since the prior release, see ["keep a changelog"]-->
+
+**Checklist**  
+
+Release numbering must follow [Semantic Versioning]. These steps assume the
+current `master` branch **development** version is **MAJOR.MINOR.0**.
+
+On the day of the feature freeze for the current development **MAJOR.MINOR.0**
+release, create a new release branch and release candidate tag:
+
+- [ ] Double check that your local `master` is up-to-date with the upstream repo.
+- [ ] Create a new branch called `release/MAJOR.MINOR` from `master`.
+- [ ] Add a tag to the `HEAD` commit in the `release/MAJOR.MINOR` branch.
+  - The tag name should be `vMAJOR.MINOR.0-RC.1`
+  - Use message "Release MAJOR.MINOR.0 RC.1".  
+  - Make sure the tag is signed, for extra safety use the explicit `--sign` flag.
+- [ ] Push the `release/MAJOR.MINOR` branch and new tag to the `bitcoindevkit/bdk` repo.
+  - Use `git push --tags` option to push the new `vMAJOR.MINOR.0-RC.1` tag.
+
+Change `master` branch version to next development **MAJOR.MINOR+1.0** release:
+
+- [ ] Switch back to the `master` branch.
+- [ ] Create a new PR branch called `bump_dev_MAJOR_MINOR+1`, eg. `bump_dev_0_18`.
+- [ ] Bump the `bump_dev_MAJOR_MINOR+1` branch to the next development MAJOR.MINOR+1 version.
+  - Change the `Cargo.toml` version value to `MAJOR.MINOR+1.0`.
+  - The commit message should be "Bump version to MAJOR.MINOR+1.0".
+- [ ] Create PR and merge the `bump_dev_MAJOR_MINOR+1` branch to `master`.
+  - Title PR "Bump dev version to MAJOR.MINOR+1.0".
+
+If any issues need to be fixed before the **MAJOR.MINOR.0** version is released:
+
+- [ ] Merge fix PRs to the `master` branch.
+- [ ] Git cherry-pick fix commits to the `release/MAJOR.MINOR` branch.
+- [ ] Verify fixes in `release/MAJOR.MINOR` branch.
+- [ ] Add a tag to the `HEAD` commit in the `release/MAJOR.MINOR` branch.
+  - The tag name should be `vMAJOR.MINOR.0-RC.x+1`, where x is the current release candidate number.
+  - Use tag message "Release MAJOR.MINOR.0 RC.x+1".  
+  - Make sure the tag is signed, for extra safety use the explicit `--sign` flag.
+- [ ] Push the new tag to the `bitcoindevkit/bdk` repo.
+  - Use `git push --tags` option to push the new `vMAJOR.MINOR.0-RC.x+1` tag.
+
+On the day of a **MAJOR.MINOR.0** release:
+
+- [ ] Add a tag to the `HEAD` commit in the `release/MAJOR.MINOR` branch.
```

Comment:
I think here you should first add a commit to bump the version and remove the rc suffix

Reply:
In the section starting on line 41 I bump the version for the next dev version in `master`.  So the `master` branch will always be at the current dev version, this is needed so someone can patch their project and point to the master branch to use the current dev branch version.

I also added a version bump to `0.22.0` in this PR to get us ready to follow this process for the upcoming feature freeze.